### ISSUE TITLE
Time Series | Bias metric

### ISSUE THREAD

**Issue by ngupta23** [pycaret/pycaret#2399]
**Is your feature request related to a problem? Please describe.**
Add a metric for Bias in the forecast 

**Describe the solution you'd like**
Several Options Exist

- [Bias, Tracking Signal, NFM](https://blog.arkieva.com/how-to-measure-bias-in-forecast/#:~:text=BIAS%20%3D%20Historical%20Forecast%20Units%20(Two,(indicates%20over%2Dforecast).)
- [CFE, CFEmin, CFEmax](https://medium.com/towards-data-science/forecast-error-measures-intermittent-demand-22617a733c9e)

**Describe alternatives you've considered**
Users can manually add them to the metrics, but it may be better if we at least provide them as options in our metrics.
Better still - add these to sktime and simply reuse.

**Comment by ngupta23** [comment]
If anyone wants to pick this up, we will be happy to accept a PR. Just be sure to discuss the issue here before starting development since there are many variants and we would need to finalize what to support. Thanks!

**Comment by kumar21120** [comment]
Hi ngupta23,

Can I pickup this issue and work on it?

**Comment by ngupta23** [comment]
Yes absolutely. Please describe the metrics you plan (specifically how bias will be calculated) to add here so we can finalize the design before you start coding. 

You can look at the existing metric container to check how it is done right now.

Thanks.

**Comment by kumar21120** [comment]
I am planning to add Cumulative Forecast Error (CFE), which is the summation of error at each period (Actual - Forecast). The negative value of CFE will represent over-forecast, whereas the positive value will represent under-forecast.

**Comment by kumar21120** [comment]
Can I go ahead with the CFE implementation, @ngupta23? or do you have any suggestions regarding the metric?

**Comment by ngupta23** [comment]
I think that sounds good. Just note that it may be better to normalize this with respect to the number of points in the forecast horizon. This way we can compare the metrics on the same model for different forecast horizons (e.g fh = 12 and fh = 24). It can also help us track how the bias is changing with forecast horizon (whether it is degrading or remaining stable).

When you do this, pay special attention to the cases where fh has gaps. for example, fh = [2, 3, 4, 10, 11, 12] should only divide CFE by 6 (since there are 6 data points in the FH) and not 12.

You may also need to add unit tests to check for these conditions. Some examples of how this is done for other "custom" metrics can be found here.

https://github.com/pycaret/pycaret/blob/9b934b0654e12ca627a737c526703dce5317aebf/tests/test_time_series_metrics.py#L19-L74 

Let me know if you need any more inputs.

Thanks!

**Comment by kumar21120** [comment]
Thank you @ngupta23

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
